### PR TITLE
Expose current pipeline in python functions

### PR DESCRIPTION
**Pull request by banasraf**
## Category:

**New feature**

## Description:

It enables python operators to access the pipeline they are being executed by with `Pipeline.current()`. 
This allows to check some useful properties, most importantly, `device_id`

### Affected modules and functionalities:

python_function

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/python/nvidia/dali/ops/_operators/python_function.py [hunk 1]
```diff
@@ -119,47 +119,52 @@ def check_outputs(outputs, num_outputs):
...
-        if arr_out is None:
-            return
-        PythonFunction.check_outputs(arr_out, num_outputs)
-        if isinstance(arr_out, tuple):
-            return tuple(map(lambda t: to_dlpack(t), arr_out))
-        else:
-            return to_dlpack(arr_out)
+    def function_wrapper_per_sample(pipeline, function, num_outputs,
+                                    from_dlpack, to_dlpack, *dlpack_inputs):
+        with pipeline:
```
Comment (banasraf) [inline dali/python/nvidia/dali/ops/_operators/python_function.py:124]: misleading diff; only `with pieline:` is added

File: dali/python/nvidia/dali/ops/_operators/python_function.py [hunk 2]
```diff
@@ -119,47 +119,52 @@ def check_outputs(outputs, num_outputs):
...
-        arr_outs = function(*arrays)
-        if arr_outs is None:
-            return
-
-        def convert_batch(batch):
-            if isinstance(batch, list):
-                return [to_dlpack(x) for x in batch]
+    def function_wrapper_batch(pipeline, function, num_outputs,
+                               from_dlpack, to_dlpack, *dlpack_inputs):
+        with pipeline:
```
Comment (banasraf) [inline dali/python/nvidia/dali/ops/_operators/python_function.py:138]: misleading diff; only `with pieline:` is added

File: dali/test/python/operator_2/test_python_function.py [hunk 3]
```diff
@@ -493,6 +493,8 @@ def test_sink():
 
 
 def func_with_side_effects(images):
+    # print('Hahahah')
```
Comment (klecki) [inline dali/test/python/operator_2/test_python_function.py:496]: :)

**Review by jantonguirao — APPROVED** [review]

File: dali/test/python/operator_2/test_python_function.py (hunk 3 above)
Comment (banasraf) [reply dali/test/python/operator_2/test_python_function.py]: removed

**Comment by banasraf** [comment]
!build

**Review by klecki — APPROVED** [review]

**Pull request merged**